byron-adapter/src/exec/Main.hs

```diff
@@ -0,0 +1,352 @@
+        nArgs
+        (configGeneratedSecrets genesisConfig)
+      putStrLn "Got node params"
+      let genesisBlock = genesisBlock0 (configProtocolMagic genesisConfig)
+                                       (configGenesisHash genesisConfig)
+                                       (genesisLeaders genesisConfig)
+          -- Here's all of the config needed for the Byron proxy server itself.
+          bpc :: ByronProxyConfig
+          bpc = ByronProxyConfig
+            { bpcAdoptedBVData = configBlockVersionData genesisConfig
+              -- ^ Hopefully that never needs to change.
+            , bpcNetworkConfig = npNetworkConfig nodeParams
+            , bpcDiffusionConfig = FullDiffusionConfiguration
+                { fdcProtocolMagic = configProtocolMagic genesisConfig
+                , fdcProtocolConstants = configProtocolConstants genesisConfig
+                , fdcRecoveryHeadersMessage = recoveryHeadersMessage
+                , fdcLastKnownBlockVersion = lastKnownBlockVersion updateConfiguration
+                , fdcConvEstablishTimeout = networkConnectionTimeout
+                -- Diffusion layer logs will have "diffusion" in their names.
+                , fdcTrace = appendName "diffusion" logTrace
+                , fdcStreamWindow = streamWindow
+                }
+              -- 40 seconds.
+            , bpcPoolRoundInterval = 40000000
+            , bpcSendQueueSize     = 1
+            , bpcRecvQueueSize     = 1
+            }
+          -- The database configuration follows.
+          -- We need to open it up before starting the Byron proxy server, since
+          -- it needs some block source that we derive from the database.
+          --
+          -- The number of slots in an epoch. We use the one in the genesis
+          -- configuration, and we assume it will never change.
+          epochSlots :: SlotCount
+          epochSlots = configEpochSlots genesisConfig
+          -- Default database path is ./db-byron-adapter
+          dbFilePath :: FsPath
+          dbFilePath = maybe ["db-byron-adapter"] pure (dbPath cArgs)
+          fsMountPoint :: MountPoint
+          fsMountPoint = MountPoint ""
+          fs :: HasFS IO
+          fs = ioHasFS fsMountPoint
+          errorHandling :: ErrorHandling ImmutableDBError IO
+          errorHandling = ErrorHandling
+            { throwError = throwIO
+            , catchError = catch
+            }
+          epochSizes :: Map Epoch EpochSize
+          -- FIXME the 'fromIntegral' casts from 'Word64' to 'Word'.
+          -- For sufficiently high k, on certain machines, there could be an
+          -- overflow.
+          -- FIXME need to be able to give a function `const epochSlots`.
+          -- 0 to 200 should be fine for now... I think.
+          --
+          -- Add 1 to the slot count because we put EBBs at the end of every
+          -- epoch.
+          epochSizes = Map.fromList (fmap (\i -> (i, fromIntegral (getSlotCount epochSlots) + 1)) [0..200])
```

Comment:
Obviously `[0..200]` is not good. Will get rid of this when there is support from immutable DB.